hosting/libs/firebase/auth.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  type User,
+  GoogleAuthProvider,
+  signInWithPopup,
+  onAuthStateChanged as _onAuthStateChanged,
+} from "firebase/auth";
+
+import { firebaseAuth } from "./config";
+
+export function onAuthStateChanged(callback: (authUser: User | null) => void) {
+  return _onAuthStateChanged(firebaseAuth, callback);
+}
+
+export async function signInWithGoogle() {
+  const provider = new GoogleAuthProvider();
+
+  try {
+    const result = await signInWithPopup(firebaseAuth, provider);
+
+    if (!result || !result.user) {
+      throw new Error("Google sign in failed");
+    }
+    return result.user.uid;
```

Comment:
Handle success condition with snackbar
```suggestion
   Snackbar.success('Login success')
    return result.user.uid;
```

Reply:
I think no need to notify user that login succeeded. Just redirect into the UI and the user-feedback of success is clear.